landlab/components/bedrock_landslider/bedrock_landslider.py

```diff
@@ -0,0 +1,692 @@
+            nb_up = nb[topo[nb] > cP_el]
+
+            X_cP = self.grid.node_x[cP]
+            Y_cP = self.grid.node_y[cP]
+
+            distToIni_all = np.sqrt(
+                np.add(
+                    np.square(X_cP - self.grid.node_x[nb_up]),
+                    np.square(Y_cP - self.grid.node_y[nb_up]),
+                )
+            )
+            all_iP_el = topo[nb_up]
+            s_slide_all = (all_iP_el - cP_el) / distToIni_all
+
+            nb_up = nb_up[s_slide_all > self._angle_int_frict]
+            s_slide_all = s_slide_all[s_slide_all > self._angle_int_frict]
+
+            if s_slide_all.size > 0:
+                s_slide = max(s_slide_all)
+                storeV_bed = np.float64(0.0)
+                storeV_sed = np.float64(0.0)
+                upstream = 0
+                uP = nb_up
+                if not self.landslides_on_boundary_nodes:
+                    uP = uP[(self.grid.node_is_boundary(uP) is False)]
+                # Fix sliding angle of particular LS
+                ang_sl = np.float64((self._angle_int_frict + s_slide) / 2)
+                stall = 0
+
+                while uP.size > 0 & (
+                    upstream <= self._max_pixelsize_landslide and stall < 1e4
+                ):
+                    # print(uP.size)
+                    distToIni = np.sqrt(
+                        np.add(
+                            np.square(X_cP - self.grid.node_x[uP[0]]),
+                            np.square(Y_cP - self.grid.node_y[uP[0]]),
+                        )
+                    )
+                    newEl = cP_el + distToIni * ang_sl
+                    stall += 1
+                    if newEl < topo[uP[0]]:
+                        # Do actual slide
+                        upstream = upstream + 1
+                        sed_LS_E = np.float64(
+                            max(0.0, min(soil_d[uP[0]], topo[uP[0]] - newEl))
+                        )
+                        soil_d[uP[0]] -= sed_LS_E
+                        topo[uP[0]] = newEl
+                        bed_LS_E = max(0, bed[uP[0]] - (newEl - soil_d[uP[0]]))
+                        bed[uP[0]] -= bed_LS_E
+                        topo[uP[0]] = newEl
+
+                        vol_sed = sed_LS_E * (1 - self._phi) * (self.grid.dx ** 2)
+                        vol_bed = bed_LS_E * (self.grid.dx ** 2)
+                        storeV_sed = storeV_sed + vol_sed
+                        storeV_bed = storeV_bed + vol_bed
+
+                        nb = np.concatenate(
+                            (
+                                self.grid.active_adjacent_nodes_at_node[uP[0]],
+                                self.grid.diagonal_adjacent_nodes_at_node[uP[0]],
+                            )
+                        )
+                        nb = nb[nb != -1]
+                        nb_up = nb[topo[nb] > cP_el]
+                        uP = [*uP, *nb_up]
+
+                        temp, idx = np.unique(uP, "first")
```

Comment:
I'm not sure why you are using the *"first"*. I think maybe you want,
```python
temp, idx = np.unique(uP, return_index=True)
```

Reply:
Yup, will do it too. Changed